### ISSUE TITLE
Please use the ms-DS-ConsistencyGuid word consistently

### ISSUE THREAD

**Issue by grimson73** [MicrosoftDocs/entra-docs#184]
Hi,
Just for Bing, Google and search optimization:
Please change msdsConsistencyGuid to mS-DS-ConsistencyGuid according to the https://learn.microsoft.com/en-us/windows/win32/adschema/a-ms-ds-consistencyguid documentation

msdsConsistencyGuid is used 3 times in the 'Merging objects from disconnected sources' article.

Thanks :)

And maybe for bonus, explain why the mS-DS-ConsistencyGuid isn't stamped on ADDS accounts anymore by Microsoft Entra Cloud Sync?

---
#### Document Details

⚠ *Do not edit this section. It is required for learn.microsoft.com ➟ GitHub issue linking.*

* ID: 793df638-faa2-8d6b-1b94-f6ac79b84368
* Version Independent ID: 40a3d7d6-8aa5-d2db-9409-c7f5fb11724d
* Content: [Microsoft Entra Cloud Sync supported topologies and scenarios - Microsoft Entra ID](https://learn.microsoft.com/en-us/entra/identity/hybrid/cloud-sync/plan-cloud-sync-topologies)
* Content Source: [docs/identity/hybrid/cloud-sync/plan-cloud-sync-topologies.md](https://github.com/MicrosoftDocs/entra-docs/blob/main/docs/identity/hybrid/cloud-sync/plan-cloud-sync-topologies.md)
* Service: **active-directory**
* Sub-service: **hybrid**
* GitHub Login: @billmath
* Microsoft Alias: **billmath**

**Comment by cnotin** [comment]
> explain why the mS-DS-ConsistencyGuid isn't stamped on ADDS accounts anymore by Microsoft Entra Cloud Sync

That's a great question and I was wondering the same thing! 

I suppose that's what Microsoft means [with](https://learn.microsoft.com/en-us/entra/identity/hybrid/cloud-sync/reference-cloud-sync-faq#does-cloud-sync-support-writeback-of-ms-ds-consistencyguid-for-any-object-):

> cloud sync doesn't support writeback of ms-ds-consistencyGUID for any object (including user objects).

For me, a writeback is when you modify something in Entra ID and it is synced back ("written back") to AD, but perhaps here it means the usual mechanism where the objectGuid is "written back" to ms-DS-ConsistencyGuid? 🤔

Even though the doc also [mentions that](https://learn.microsoft.com/en-us/entra/identity/hybrid/cloud-sync/reference-cloud-sync-faq#can-i-change-the-source-anchor-for-cloud-sync-):

> cloud sync uses ms-ds-consistency-GUID with a fallback to ObjectGUID as source anchor

So, is it expected that customers who migrate users from domains (case when it's important to use `ms-DS-ConsistencyGuid` instead of `objectGUID`) to manually fill `ms-DS-ConsistencyGuid`? 🤔🤔 That's what [this reply](https://github.com/MicrosoftDocs/azure-docs/issues/70087#issuecomment-777394876) from @ssivas implies...

> Cloud sync does not support writeback of ms-ds-consistencyGUID for any object. Unlink traditional AD connect, for any given on-premises AD User object whose ms-DS-ConsistencyGuid attribute isn't populated, Azure AD Connect writes its objectGUID value back to the ms-DS-ConsistencyGuid attribute in on-premises Active Directory. After the ms-DS-ConsistencyGuid attribute is populated, Azure AD Connect then exports the object to Azure AD.
> For inter forest object move, the ms-DS-ConsistencyGuid attribute needs to be manually populated during migration. For example: You would create the bbb\user in the forest they are moving to then copy over the objectGUID value from aaa\user to bbb\user's ms-DS-ConsistencyGuid attribute in on-premises Active Directory).


So to summarize, I'm really looking forward to a definitive answer and a doc update accordingly. Many thanks!

**Comment by grimson73** [comment]
Thanks for chiming in and pointing to an earlier topic about this :)

Digging further I found this:

**Things to remember about all scenarios and topologies**
[https://learn.microsoft.com/en-us/entra/identity/hybrid/cloud-sync/plan-cloud-sync-topologies#things-to-remember-about-all-scenarios-and-topologies]

> The information below should be kept in mind, when selecting a solution.
> 
> - Users and groups must be uniquely identified across all forests
> - Matching across forests doesn't occur with cloud sync
> - The source anchor for objects is chosen automatically. It uses ms-DS-ConsistencyGuid if present, otherwise ObjectGUID is used.
Guess this states that the ms-DS-ConsistencyGuid isn't populated anymore. Objects with the ms-DS-ConsistencyGuid will be used but no more ms-DS-ConsistencyGuid stamping on new objects. 

> - You can't change the attribute that is used for source anchor.

So this seems to be some sort of answer, the start of deprecating the ms-DS-ConsistencyGuid attribute.

**Comment by SaibabaBalapur-MSFT** [comment]
@billmath
Can you please check and add your comments on this doc update request as applicable.

@cnotin  @grimson73  
I'm going to assign this to the document author so they can take a look at it accordingly.

**Comment by cnotin** [comment]
Interesting, thanks for digging further!

> So guess only the 'inter forest move' does 'suffer' from this reverting behavior?

Inter-domain too. The usage of ms-DS-ConsistencyGuid as source anchor was introduced because this attribute doesn't change when the object is migrated to a different domain/forest, contrary to objectGUID (which changes and can only be set by AD)



> Potential security thingy?

https://www.semperis.com/blog/syncjacking-azure-ad-account-takeover/ ;)

**Comment by billmath** [comment]
@grimson73 Thanks for catching this.  Hopefully this helps.

The sourceAnchor attribute is defined as an attribute immutable during the lifetime of an object. It uniquely identifies an object as being the same object on-premises and in Microsoft Entra ID. The attribute is also called immutableId and the two names are used interchangeably

By default, Azure AD Connect (version 1.1.486.0 and older) uses objectGUID as the Source Anchor for on-premises AD objects. Because objectGuid is system-generated, you cannot set its value when creating an AD object. There are some scenarios where you may be required to set the Source Anchor attribute to specific values. 

For this reason, an option was made available to allow customers to change the source anchor to a non-system generated attribute.  ms-DS-ConsistencyGuid was selected as the attribute to use.  https://learn.microsoft.com/en-us/entra/identity/hybrid/connect/plan-connect-design-concepts#sourceanchor

When cloud sync was introduced, it went the other direction.  Use ms-DS-ConsistencyGuid first and then fallback to ObjectGUID.

In order to create the user object in Microsoft Entra ID, cloud sync needs the source anchor with a value in it.  However, if ms-DS-ConsistencyGuid is not set or null, it will use the OjbectGUID (which is always present).   But it will not writeback that value to ms-DS-ConsistencyGuid.  This differs from Connect Sync, which will writeback the ObjectGuid value to ms-DS-ConsistencyGuid if Connect Sync has been setup to use ms-DS-ConsistencyGuid as the anchor and it has no value.  Cloud sync never wrote the ms-DS-ConsistencyGuid back.

Sorry for in advance for the inconsistency in the spelling (this is a bad habit based on our use of the word as an actual attribute and the regular word).  Although it appears differently is areas, it should be ms-DS-ConsistencyGuid, which is how it appears on user accounts in Active Directory.  

I hope the above explains more and I have updated all of the instnaces to be ms-DS-ConsistencyGuid.  Thanks for catching this.

#please-close

**Comment by cnotin** [comment]
Thank you for the clarification that Entra Cloud Sync doesn't writeback contrary to Entra Connect! Very helpful 🙇

**Comment by grimson73** [comment]
Thanks @billmath for this valuable information and effort to correct maintain the documentation. Happy to see feedback is given on simple questions :)
